tests/end-to-end/backup-restore-test/backupe2e/service-catalog/helm_broker.go

```diff
@@ -0,0 +1,317 @@
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	bu "github.com/kyma-project/kyma/components/service-binding-usage-controller/pkg/client/clientset/versioned"
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/go-redis/redis"
+	"fmt"
+	"github.com/pkg/errors"
+	"os"
+	"k8s.io/client-go/tools/clientcmd"
+	. "github.com/smartystreets/goconvey/convey"
+
+	"time"
+)
+
+const (
+	redisInstanceName     = "redis"
+	redisBindingName      = "redis-credentials"
+	redisBindingUsageName = "redis-bu"
+
+	// key/values used to store in the Redis DB
+	sampleKey = "key-001"
+	sampleVal = "val-001"
+)
+
+func NewHelmBrokerTest() (HelmBrokerTest, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+	if err != nil {
+		return HelmBrokerTest{}, err
+	}
+	k8sCS, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return HelmBrokerTest{}, err
+	}
+	scCS, err := sc.NewForConfig(config)
+	if err != nil {
+		return HelmBrokerTest{}, err
+	}
+	buSc, err := bu.NewForConfig(config)
+	if err != nil {
+		return HelmBrokerTest{}, err
+	}
+
+	return HelmBrokerTest{
+		BUInterface:             buSc,
+		ServiceCatalogInterface: scCS,
+		K8sInterface:            k8sCS,
+	}, nil
+}
+
+type HelmBrokerTest struct {
+	ServiceCatalogInterface clientset.Interface
+	K8sInterface            kubernetes.Interface
+	BUInterface             bu.Interface
+}
+
+type helmBrokerFlow struct {
+	brokersFlow
+
+	scInterface clientset.Interface
+	buInterface bu.Interface
+}
+
+func (t HelmBrokerTest) CreateResources(namespace string) {
+	t.newFlow(namespace).createResources()
+}
+
+func (t HelmBrokerTest) TestResources(namespace string) {
+	t.newFlow(namespace).testResources()
+}
+
+func (t HelmBrokerTest) DeleteResources(namespace string) {
+	t.newFlow(namespace).deleteResources()
+}
+
+func (t *HelmBrokerTest) newFlow(namespace string) *helmBrokerFlow {
+	return &helmBrokerFlow{
+		brokersFlow: brokersFlow{
+			namespace: namespace,
+			log:       logrus.WithField("test", "helm-broker"),
+
+			buInterface:  t.BUInterface,
+			scInterface:  t.ServiceCatalogInterface,
+			k8sInterface: t.K8sInterface,
+		},
+		buInterface: t.BUInterface,
+		scInterface: t.ServiceCatalogInterface,
+	}
+}
+
+func (f *helmBrokerFlow) createResources() {
+	for _, fn := range []func() error{
+		f.createRedisInstance,
+		f.deployEnvTester,
+		f.waitForEnvTester,
+		f.waitForRedisInstance,
+		f.createRedisBindingAndWaitForReadiness,
+		f.createRedisBindingUsageAndWaitForReadiness,
+	} {
+		err := fn()
+		if err != nil {
+			f.logReport()
+		}
+		So(err, ShouldBeNil)
+	}
+}
+
+func (f *helmBrokerFlow) testResources() {
+	// iterate over testing steps
+	for _, fn := range []func() error{
+		f.storeKeyInRedis,
```

Comment:
as discussed, we should check if value exists there, 

additionaly, you can try to store new entry to check if connection is still working properly.

Reply:
We are checking if the connection is still working properly after restore by getting value from it so I decided not to add another assertion.